stan/math/rev/core/count_vars.hpp

```diff
@@ -0,0 +1,157 @@
+#ifndef STAN_MATH_REV_CORE_COUNT_VARS_HPP
+#define STAN_MATH_REV_CORE_COUNT_VARS_HPP
```

Comment:
Why is this in core? Shouldn't it be in fun (or maybe meta?)

Reply:
I just put it in core cause it's explicitly dealing with things with vars, it doesn't have use outside of var specializations. I didn't put it in meta cause it's a runtime thing (dynamically sized arrays).